tokio-io/src/split.rs

```diff
@@ -168,4 +228,26 @@ mod tests {
             ok::<(), ()>(())
         })).unwrap();
     }
+
+    #[test]
+    fn unsplit_ok() {
```

Comment:
Should we also add a test that reading or writing from an `unsplit` IO actually reads/writes to/from the same destination as the pre-split IO?